indico/modules/receipts/util.py

```diff
@@ -145,22 +148,58 @@ def compile_jinja_code(code: str, template_context: dict, *, use_stack: bool = F
         raise UnprocessableEntity(e)
 
 
-def sandboxed_url_fetcher(event: Event, allow_event_logo: bool = False) -> t.Callable[[str], dict]:
+def sandboxed_url_fetcher(event: Event, allow_event_images: bool = False) -> t.Callable[[str], dict]:
     """Fetch also "event-local" URLs.
 
     More info on fetchers: https://doc.courtbouillon.org/weasyprint/stable/first_steps.html#url-fetchers
     """
     allow_external_urls = receipts_settings.get('allow_external_urls')
 
     def _fetcher(url: str) -> dict:
-        if allow_event_logo and url == 'event://logo':
-            return {
-                'mime_type': event.logo_metadata['content_type'],
-                'string': event.logo
-            }
+        url_data = urlparse(url)
+        if allow_event_images and url_data.scheme == 'event':
+            if url_data.netloc == 'logo':
+                return {
+                    'mime_type': event.logo_metadata['content_type'],
```

Comment:
This currently (also in production) causes a TypeError when the event has no logo. It's silent (just ends up being logged to sentry), but maybe you could add a `raise ValueError('Event has no logo')` if the event has no logo (you can use `event.has_logo` to check that)?

It's not directly related to this PR but since you're already changing a lot in this function, it'd be easier to include it there than me having to fix it after merging your PR  (or doing it before and making you hate me for adding rebase conflicts ;))